### PR TITLE
Document well defined eval order starting from Move 2

### PR DESCRIPTION
**Pull request by vineethk**
### Description

Document the well specified evaluation order starting from Move 2.

### Checklist

- Do all Lints pass?
  - [x] Have you ran `pnpm spellcheck`?
  - [x] Have you ran `pnpm fmt`?
  - [x] Have you ran `pnpm lint`?

### CODE REVIEW

**Comment by vineethk** [comment]
* **#704** <a href="https://app.graphite.dev/github/pr/aptos-labs/developer-docs/704?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @vineethk and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by rahxephon89 — APPROVED** [review]

File: apps/nextra/pages/en/build/smart-contracts/book/move-2.mdx [hunk 1]
```diff
@@ -12,15 +12,15 @@ The Move 2.1 language release adds the following features to Move:
...
+## Move 2.0
 
 The Move 2.0 language release adds the following features to Move:
 
 - **Enum Types** add the option to define different variants of data layout in one storable type. They are documented in the [Enum Type section](enums.mdx).
 
 - **Receiver Style Functions** add the ability to call functions in the familiar notation `value.func(arg)`. They are documented in [this section](functions.mdx#dot-receiver-function-call-style).
 
-- **Index Notation** allows to access [elements of vectors](vector.mdx#index-notation-for-vectors) and of [resource storage](global-storage-operators.mdx#index-notation-for-storage-operators) with notations like `&mut vector[index]`, or `&mut Resource[addr]`, respectively. 
+- **Index Notation** allows to access [elements of vectors](vector.mdx#index-notation-for-vectors) and of [resource storage](global-storage-operators.mdx#index-notation-for-storage-operators) with notations like `&mut vector[index]`, or `&mut Resource[addr]`, respectively.
```
Comment (brmataptos) [inline apps/nextra/pages/en/build/smart-contracts/book/move-2.mdx:23]: "allows to access" is pretty weird.  Suggest "allows access to" instead.

**Review by brmataptos — APPROVED** [review]
Suggest a change in some existing line, other than that is fine.

File: apps/nextra/pages/en/build/smart-contracts/book/move-2.mdx (hunk 1 above)
Comment (vineethk) [reply apps/nextra/pages/en/build/smart-contracts/book/move-2.mdx]: done

**Pull request merged**